src/main/java/me/pheric/pcore/database/Database.java

```diff
@@ -0,0 +1,171 @@
+package me.pheric.pcore.database;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Manages and connects to a MySQL database
+ *
+ * @author StealthMajr (stealthmajr@gmail.com)
+ * @see Connection
+ * @see HikariDataSource
+ * @see Result
+ * @see DB
+ * @since 0.1.0
+ */
+public class Database {
+
+    private HikariDataSource dataSource;
+
+    /**
+     * Initializes the class
+     *
+     * @param jdbcUrl  The JDBC URL for connecting, usually in the following format:
+     *                 jdbc:mysql://hostname:port/database
+     * @param username The username of this MySQL connection
+     * @param password The password of this MySQL connection
+     */
+    public Database(String jdbcUrl, String username, String password) {
+        dataSource = new HikariDataSource();
+        dataSource.setDriverClassName("com.mysql.jdbc.Driver");
+        dataSource.setJdbcUrl(jdbcUrl);
+        dataSource.setUsername(username);
+        dataSource.setPassword(password);
+        dataSource.setPoolName("Core");
+
+        // Configuration settings for the HikariDataSource
+        dataSource.setMaximumPoolSize(10);
+    }
+
+    /**
+     * Updates a database row with parameters
+     *
+     * @param query  The update statement
+     * @param params The parameters to sanitize in the query
+     * @return Returns -1 if there was an error, 0 if nothing was updated and vice-versa for 1
+     */
+    int update(String query, Object... params) {
+        // Try-resources block to automatically close the connection and prepared statement
+        try (Connection connection = getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            // Loops through the parameters array and sets it on the prepared statement
+            for (int i = 0; i < params.length; i++) {
+                // Set the object in the array to the prepared statement
+                // Which pretty much replaces the quotations
+                // in MySQL queries based on index
+                preparedStatement.setObject((i + 1), params[i]);
+            }
+            // Execute the update and return the value
+            return preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            // Indicates an error
+            return -1;
```

Comment:
This doesn't help much with debugging. Consider throwing an exception or printing a stack trace (maybe add a debug option that gets disabled in prod, or have the stack trace outputted to a separate logger)

Reply:
I had a logger there but removed it because there wasn't an obtainable one in PCore.